v2/emailpassword/common-customizations/usermetadata/store-data.mdx

```diff
@@ -0,0 +1,279 @@
+---
+id: store-data
+title: Store data
+hide_title: true
+---
+
+<!-- COPY DOCS -->
+<!-- ./emailpassword/common-customizations/usermetadata/store-data.mdx -->
+
+import BackendSDKTabs from "/src/components/tabs/BackendSDKTabs";
+import TabItem from "@theme/TabItem";
+import NpmOrScriptTabs from "/src/components/tabs/NpmOrScriptTabs";
+import NodeJSFrameworkSubTabs from "/src/components/tabs/NodeJSFrameworkSubTabs";
+
+# Storing data about a user
+
+You can store/update data about a user through the `UserMetadata.updateUserMetadata` method using their `userId`.
+
+## How the metadata updates work
+
+We update the metadata using the following steps:
+
+1. Load old metadata from DB or use an empty object if we have nothing stored.
+2. Overwrite all root-level properties with the props of the update object.
+3. Remove all root-level properties with a null value.
+
+This way, you can update parts of the metadata object without loading and merging the whole thing in the backend code.
+
+:::important
+Only root-level properties are merged into the stored object. Nested objects and all lower-level properties will be replaced.
+:::
+
+### Example
+
+- The stored object has a theme set in preferences, emails enabled in notifications, and a single todo item, telling them to switch to text messages:
+
+```json
+{
+  "preferences": { "theme": "dark" },
+  "notifications": { "email": true },
+  "todos": ["use-text-notifs"]
+}
+```
+
+- Now, we want to update this by changing the notification setting and removing the entire todo list:
+
+```json
+{
+  "notifications": { "sms": true },
+  "todos": null
```

Comment:
Passing null to the metadata object only works for node SDK. Other SDKs don't have a way to differentiate between a missing value vs `null` value. So this section should probably only talk about adding and updating data and not removing it. Then in the clear metadata section, we can talk about removing a particular field along with clearing the full metadata object